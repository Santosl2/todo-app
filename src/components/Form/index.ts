export * from "./Input";
export * from "./MainForm";
export * from "./Button";
