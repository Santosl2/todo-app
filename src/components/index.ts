export * from "./Header";
export * from "./Form";
export * from "./Badge";
export * from "./TextWithBadge";
export * from "./Item";
export * from "./NoResults";
