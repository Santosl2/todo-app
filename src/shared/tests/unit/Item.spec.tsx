/* eslint-disable testing-library/no-unnecessary-act */
/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable unused-imports/no-unused-imports */

import { act } from "react-dom/test-utils";

import { Item } from "@/components";
import { fireEvent, screen } from "@testing-library/react";

import { renderWithRedux } from "../utils/CustomRender";

// jest.mock("react-redux", () => ({
//   ...jest.requireActual("react-redux"),
//   useSelector: jest.fn(),
//   useDispatch: jest.fn(),
// }));

// const mockedSelectorTodos = jest.spyOn(
//   require("../../hooks/useSelectorTodos"),
//   "useSelectorTodos"
// );

const initialState = {
  todos: [
    {
      id: 1,
      value: "Test Task",
      isCompleted: false,
    },
  ],
};

function render() {
  const { mockStore, rerenderWithRedux } = renderWithRedux(
    <Item id={1} text="Test Task" />,
    initialState
  );

  return { mockStore, rerenderWithRedux };
}

describe("Item", () => {
  it("should render correctly", () => {
    render();

    expect(screen.getByText("Test Task")).toBeInTheDocument();
  });

  it("should be able to change Todo to completed", async () => {
    act(() => {
      render();
    });

    const todoCheckBox = screen.getByTestId(
      "checkboxItemTest"
    ) as HTMLInputElement;

    fireEvent.click(todoCheckBox);

    expect(screen.getByText("Test Task")).toHaveClass("line-through");

    expect(screen.getByText("Test Task")).toHaveClass("text-gray-300");
  });

  it("should be able to delete Todo", async () => {
    act(() => {
      render();
    });

    const deleteBtn = screen.getByTestId("deleteItemTest") as HTMLInputElement;

    fireEvent.click(deleteBtn);

    expect(screen.queryByText("Test Task")).not.toBeInTheDocument();
  });

  it("match snapshot", () => {
    render();
    expect(screen.getByText("Test Task")).toMatchSnapshot();
  });
});
