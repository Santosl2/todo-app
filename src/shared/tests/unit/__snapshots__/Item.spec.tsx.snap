// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item match snapshot 1`] = `
<p
  class="text-sm text-gray-100"
>
  Test Task
</p>
`;
